db/error_handler_fs_test.cc

```diff
@@ -1827,10 +1827,95 @@ TEST_F(DBErrorHandlingFSTest, FLushWritNoWALRetryableErrorAutoRecover1) {
   Destroy(options);
 }
 
-TEST_F(DBErrorHandlingFSTest, FLushWritNoWALRetryableErrorAutoRecover2) {
+TEST_F(DBErrorHandlingFSTest, MultipleRecoveryThreads) {
```

Comment:
Is it possible to reduce the usage of sync point (both in the core code and in the test)? 

It's likely that we will refactor ErrorHandler to reduce the possibility of new race condition. More sync point will make it harder to refactor.

Reply:
I removed one of the three SyncPoints as it's no longer necessary, but the first two are still needed to ensure we can create the race condition.

I do agree that more sync points make it harder to change without breaking tests, etc. When we refactor, some of these sync points may not make sense and we may have to rewrite the tests. I cannot think if a concrete example yet, though.